blinding.go

```diff
@@ -0,0 +1,169 @@
+package sphinx
+
+import (
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+)
+
+const routeBlindingHMACKey = "blinded_node_id"
+
+// BlindedPath represents all the data that the creator of a blinded path must
+// transmit to the builder of route that will send to this path.
+type BlindedPath struct {
+	// IntroductionPoint is the real node ID of the first hop in the blinded
+	// path. The sender should be able to find this node in the network
+	// graph and route to it.
+	IntroductionPoint *btcec.PublicKey
+
+	// BlindingPoint is the first ephemeral blinding point. This is the
+	// point that the introduction node will use in order to create a shared
+	// secret with the builder of the blinded route. This point will need
+	// to be communicated to the introduction node by the sender in some
+	// way.
+	BlindingPoint *btcec.PublicKey
+
+	// BlindedHops is a list of ordered BlindedHopInfo. Each entry
+	// represents a hop in the blinded path along with the encrypted data to
+	// be sent to that node. Note that the first entry in the list
+	// represents the introduction point of the path and so the node ID of
+	// this point does not strictly need to be transmitted to the sender
+	// since they will be able to derive the point using the BlindingPoint.
+	BlindedHops []*BlindedHopInfo
+}
+
+// hopInfo represents a single hop in a blinded path. The BlindedHopInfo wrapper
+// of this type should be used when the members of the struct represent the real
+// node pub key and clear text payload. The UnBlindedHopInfo wrapper should be
+// used when the blinded node pub key and payload are being used.
+type hopInfo struct {
+	// NodePub is the public key of the hop. It represents the real pub key
+	// if the UnBlindedHopInfo wrapper is used and represents the blinded
+	// pub key if the BlindedHopInfo wrapper is used.
+	NodePub *btcec.PublicKey
+
+	// Payload is the data to be transported to the hop. It is the cleartext
+	// if the UnBlindedHopInfo wrapper is used and is the ciphertext if the
+	// BlindedHopInfo wrapper is used.
+	Payload []byte
+}
+
+// BlindedHopInfo represents a blinded node pub key along with the blinded data
+// for that node.
+type BlindedHopInfo hopInfo
+
+// UnBlindedHopInfo represents a real node pub key along with the plaintext data
+// for that node.
+type UnBlindedHopInfo hopInfo
+
+// BuildBlindedPath creates a new BlindedPath from a list of UnBlindedHopInfo
+// and a session key.
+func BuildBlindedPath(sessionKey *btcec.PrivateKey,
+	paymentPath []*UnBlindedHopInfo) (*BlindedPath, error) {
+
+	if len(paymentPath) < 1 {
+		return nil, fmt.Errorf("at least 1 hop is required to create " +
```

Comment:
Don't need `Errorf` with no params

Reply:
there is no `fmt.Error(` option. `fmt.Errorf` is used throughout the code base even when there are no params